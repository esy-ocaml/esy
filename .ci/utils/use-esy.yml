# steps to install esy globally

steps:
- script: "yarn global add @esy-nightly/esy@0.6.8-bbdfb8"
  displayName: "Install esy (via yarn)"
  condition: ne(variables['AGENT.OS'], 'Windows_NT')
  
- script: |
    yarn config set prefix C:\
    yarn global add @esy-nightly/esy@0.6.8-bbdfb8
  displayName: "Install esy (via yarn)"
  condition: eq(variables['AGENT.OS'], 'Windows_NT')

- bash: |
    YARN_BIN_DIR=$(yarn global bin)
    echo $YARN_BIN_DIR
    echo "##vso[task.setvariable variable=YARN_BIN_DIR]$YARN_BIN_DIR"
  displayName: "Setting $YARN_BIN_DIR"

- bash: |
    echo "##vso[task.setvariable variable=PATH]$PATH:$YARN_BIN_DIR" 
  displayName: "Ensure yarn install global tools are available on $PATH"
  condition: ne(variables['AGENT.OS'], 'Windows_NT')

- powershell: Write-Host "##vso[task.setvariable variable=PATH;]${env:PATH};${env:YARN_BIN_DIR}"
  displayName: "Ensure yarn install global tools are available on $PATH"
  condition: eq(variables['AGENT.OS'], 'Windows_NT')
