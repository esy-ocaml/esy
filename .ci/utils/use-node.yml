# steps to use node on agent

steps:
  - task: NodeTool@0
    displayName: "Use Node 15.x"
    inputs:
      versionSpec: 15.x
