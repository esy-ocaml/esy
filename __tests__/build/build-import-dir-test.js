/**
 * @flow
 */

import {defineTestCaseWithShell} from './utils';

defineTestCaseWithShell(
  'import-archive',
  `
    run esy build
    run esy export-dependencies
    (cd _export && tar xzf dep-1.0.0.tar.gz && rm dep-1.0.0.tar.gz)
    run ls _export

    # drop global store
    rm -rf ../esy

    DEBUG='esy:*' run esy build
  `,
);
