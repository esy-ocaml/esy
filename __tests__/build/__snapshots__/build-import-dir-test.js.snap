// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`build import-archive 1`] = `
"RUNNING: esy config ls
esy:bin using <sandbox>/.esyrc
esy:bin prefix: <testRoot>/esy
RUNNING: esy build
esy:bin using <sandbox>/.esyrc
esy:bin prefix: <testRoot>/esy
esy:bin needRebuildTarget <sandbox>/node_modules/.cache/_esy/build/bin/build
esy:bin trying to acquire lock
esy:bin:esyBuild execute
esy:build:dep@1.0.0 start building
esy:build:dep@1.0.0 testing for import: _export/dep-1.0.0
esy:build:dep@1.0.0 testing for import: _export/dep-1.0.0.tar.gz
esy:build:dep@1.0.0 build success
esy:bin trying to acquire lock
esy:bin using <sandbox>/.esyrc
esy:bin prefix: <testRoot>/esy
esy:build-dependencies checking if dependencies are built
RUNNING: esy export-dependencies
esy:bin using <sandbox>/.esyrc
esy:bin prefix: <testRoot>/esy
esy:bin using <sandbox>/.esyrc
esy:bin prefix: <testRoot>/esy
esy:bin using <sandbox>/.esyrc
esy:bin prefix: <testRoot>/esy
RUNNING: ls _export
RUNNING: rm -rf ../esy
RUNNING: esy build
esy:bin using <sandbox>/.esyrc
esy:bin prefix: <testRoot>/esy
esy:bin needRebuildTarget <sandbox>/node_modules/.cache/_esy/build/bin/build
esy:bin trying to acquire lock
esy:bin using <sandbox>/.esyrc
esy:bin prefix: <testRoot>/esy
esy:build-dependencies checking if dependencies are built
esy:bin:esyBuild execute
esy:build:dep@1.0.0 start building
esy:build:dep@1.0.0 testing for import: _export/dep-1.0.0
esy:build:dep@1.0.0 importing build: _export/dep-1.0.0
esy:build:dep@1.0.0 build success"
`;
